.github/workflows/end-to-end.yml

```diff
@@ -48,26 +49,59 @@ jobs:
           ONLY_ACTIVE_ARCH=NO \
         | tee xcodebuild.log
 
-    - name: Install Maestro
-      run: |
-        export MAESTRO_VERSION=1.36.0; curl -Ls "https://get.maestro.mobile.dev" | bash
+    - name: Store Binary
+      uses: actions/upload-artifact@v4
+      with:
+        name: duckduckgo-ios-app
+        path: DerivedData/Build/Products/Debug-iphonesimulator/DuckDuckGo.app
+    
+    - name: Upload logs when workflow failed
+      uses: actions/upload-artifact@v4
+      if: failure() || cancelled()
+      with:
+        name: BuildLogs
+        path: |
+          xcodebuild.log
+          DerivedData/Logs/Test/*.xcresult
+        retention-days: 7
 
-    - name: Release tests
-      run: |
-        export PATH="$PATH":"$HOME/.maestro/bin"; maestro cloud --apiKey ${{ secrets.MAESTRO_CLOUD_API_KEY }} -e ONBOARDING_COMPLETED=true --fail-on-timeout=true --fail-on-cancellation=true --timeout=150 --ios-version=17 --include-tags=release DerivedData/Build/Products/Debug-iphonesimulator/DuckDuckGo.app .maestro/
+  end-to-end-tests:
```

Comment:
Currently I have setup the concurrency matrix in this way

```
strategy:
      matrix:
        test-tag: [release, privacy, securityTest, adClick]
      max-parallel: 1 # Uncomment this line to run tests sequentially. 
      fail-fast: false
```

and run 

`export PATH="$PATH":"$HOME/.maestro/bin"; maestro cloud --apiKey ${{ secrets.MAESTRO_CLOUD_API_KEY }} -e ONBOARDING_COMPLETED=true --fail-on-timeout=true --fail-on-cancellation=true --timeout=150 --ios-version=17 --include-tags=${{ matrix.test-tag }} DerivedData/Build/Products/Debug-iphonesimulator/DuckDuckGo.app .maestro/`

If we decide to have run onboarding tests in CI we can change this to

  ```
  strategy:
      matrix:
        test-tag: [release, privacy, securityTest, adClick]
        include:
          - test-tag: release
            onboarding-completed: true
          - test-tag: privacy
            onboarding-completed: true
          - test-tag: securityTest
            onboarding-completed: true
          - test-tag: adClick
            onboarding-completed: true

```

and use variable `${{ matrix.onboarding-completed }}` as below 

`export PATH="$PATH":"$HOME/.maestro/bin"; maestro cloud --apiKey ${{ secrets.MAESTRO_CLOUD_API_KEY }} -e ONBOARDING_COMPLETED=${{ matrix.onboarding-completed }}--fail-on-timeout=true --fail-on-cancellation=true --timeout=150 --ios-version=17 --include-tags=${{ matrix.test-tag }} DerivedData/Build/Products/Debug-iphonesimulator/DuckDuckGo.app .maestro/`

Reply:
Actually there’s no even need to modify the matrix because the onboarding tests overrides ONBOARDING_COMPLETED env variable in their flow 🎉

Reply:
I didn't have time to test it out properly today, but the overall approach looks good to me 👍